dockerfiles/theia/Dockerfile

```diff
@@ -0,0 +1,28 @@
+# Copyright (c) 2018 Red Hat, Inc.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+
+FROM node:8-alpine
+# build dependencies requireed to compile a custom Theia
+RUN apk add --no-cache make gcc g++ python git openssh bash
+WORKDIR /home/theia
+# build Theia with all extensions to persist yarn cache in the image and
+# have default Theia build in the workspace in case no plugins are requested
+ADD https://raw.githubusercontent.com/theia-ide/theia-apps/master/theia-full-docker/latest.package.json /home/theia/package.json
+ADD theia-default-package.json /home/default/theia/package.json
+ADD src/main.js /theia_launcher/theia_launcher.js
+RUN yarn && \
+    yarn theia build && \
+    rm -rf * && \
+    cd /home/default/theia && \
+    yarn && \
+    yarn theia build
+EXPOSE 3000
+ENV USE_LOCAL_GIT=true \
+    GITHUB_TOKEN=957257352ad57b7853c8b8714d2201bf4d337b5e
```

Comment:
I do not think that this is a good idea to put someone's token in dockerfile, I think it user should pass it's own GH token if that is necessary

Reply:
yes

Reply:
Sure it's not good. But as this point we don't have any way to get user's token

Reply:
https://github.com/roblourens/vscode-ripgrep/issues/4

Reply:
What about https://docs.docker.com/engine/reference/builder/#environment-replacement?

Reply:
@l0rd how does that help? The scenario is as follows:

1. Theia image starts, we have a running container
2. A node script checks for Theia plugins env and once found runs `yarn` and `yarn theia build` against a generated or existing package.json.  It is during this build that a token is required.

Maybe I don't understand smth about env replacement.

Reply:
@eivantsov sorry I meant https://docs.docker.com/engine/reference/commandline/run/#set-environment-variables--e-env-env-file

Reply:
if you need toked in runtime  just pass it with docker run, if you need token for `docker build` you can do 
```
+ARG GITHUB_TOKEN
+ENV GITHUB_TOKEN ${GITHUB_TOKEN}
```

docker build --build-arg GITHUB_TOKEN=<token>

Reply:
A user will just launch a workspace. It is us who do docker run, Che server I mean.

Reply:
ahh, I thought it is image for CI, because all images we have in `dockerfiles/` are for CLI and they are supposed to be built on CI. if that is image for WS maybe it should be moved to github.com/eclipse/che-dockerfiles ?